File: cplusplus/design_pattern/MediatorProj/Colleague.cpp
#include "stdafx.h"
#include "Colleague.h"
#include "Mediator.h"


Colleague::Colleague()
{
}


Colleague::~Colleague()
{
}


Colleague::Colleague(std::shared_ptr<Mediator> mediator):mediator(mediator)
{
}


void Colleague::change()
{
    //mediator->colleagueChanged();
}

File: Java/pattern/MediatorProj/src/com/pattern/Colleague.java
package com.pattern;

public abstract class Colleague {
	public Mediator mediator;
	
	public Colleague(Mediator mediator) {
		super();
		this.mediator = mediator;
	}
	
	public abstract void action();
	
	public void change() {
		mediator.colleagueChanged(this);
	}
}
